internal/allocation/allocation_manager.go

```diff
@@ -244,3 +246,75 @@ func (m *Manager) GrantPermission(sourceAddr net.Addr, peerIP net.IP) error {
 
 	return errAdminProhibited
 }
+
+// AddTCPConnection creates a new outbound TCP Connection and returns the Connection-ID
+// if it succeeds.
+func (m *Manager) AddTCPConnection( // nolint: cyclop
+	allocation *Allocation,
+	peerAddress proto.PeerAddress,
+) (proto.ConnectionID, error) {
+	if len(peerAddress.IP) == 0 || peerAddress.Port == 0 {
+		return 0, errInvalidPeerAddress
+	}
+
+	rand64, err := randutil.CryptoUint64()
+	if err != nil {
+		return 0, err
+	}
+
+	conn, err := net.DialTCP("tcp4", nil, &net.TCPAddr{IP: peerAddress.IP, Port: peerAddress.Port}) // nolint: noctx
```

Comment:
This is the critical point: (1) the Dialer should be an abstract interface and (2) the local address:port must be the transport relay address for the allocation (the TCPAddr we create an `allocation.CreateAllocation` once we change that to call `AllocateConn` per the requested transport). The former can be taken care of later, but for the latter we will need to add the `SO_REUSEADDR` (and maybe also the `SO_REUSEPORT`? I always forget the difference) otherwise we cannot bind multiple peer connections to the same local addr:port. AFAIK `SO_REUSEADDR` is supported in all major OSes but `SO_REUSEPORT` is Linux only